### PR TITLE
Feat/tomb v3 stake options

### PR DESCRIPTION
**Pull request by samholmes**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202126232980932

### CODE REVIEW

**Review by Jon-edge — CHANGES_REQUESTED** [review]

File: src/plugins/stake-plugins/stakePolicy.js [hunk 1]
```diff
@@ -19,11 +20,16 @@ export type StakePolicyInfo = {
   mustClaimRewards: boolean
 }
 
-export const withGeneratedStakePolicyId = (policyInfo: StakePolicyInfo): StakePolicyInfo => {
-  // Generate a unique deterministic ID for the policy
+// Generate a unique deterministic ID for the policy
+const deriveStakePolicyId = (policyInfo: StakePolicyInfo): string => {
+  const { exchangeProvider = '' } = policyInfo
   const stakePart = policyInfo.stakeAssets.map(asset => `${asset.pluginId}:${asset.tokenId}`).join('+')
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/plugins/stake-plugins/stakePolicy.js:26]: I'll sort the array deterministically.

File: src/plugins/stake-plugins/policies/cemeteryPolicy.js [hunk 2]
```diff
@@ -103,12 +95,19 @@ export const makeCemeteryPolicy = (options: CemeteryPolicyOptions): StakePluginP
       const policyInfo = pluginInfo.policyInfo.find(p => p.stakePolicyId === stakePolicyId)
       if (policyInfo == null) throw new Error(`Stake policy '${stakePolicyId}' not found`)
 
+      const requestAsset = policyInfo.stakeAssets.find(asset => asset.tokenId === request.tokenId)
+      if (requestAsset == null) throw new Error(`Asset '${request.tokenId}' not found in policy '${stakePolicyId}'`)
+
+      const tokenACurrencyCode = policyInfo.stakeAssets[0].tokenId
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/plugins/stake-plugins/policies/cemeteryPolicy.js:102]: Discovery to do: does the order of the token address when calling addLiquidity (or similar) impact the contract call such that an incorrect amount of tokenA/B is added to the pool or if the pool is different in any way (different LP token), or not.

File: src/util/exchangeHelpers.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// @flow
+
+import { EDGE_CONTENT_SERVER } from '../constants/WalletAndCurrencyConstants.js'
```
[the remark this answers is not in the document]
Comment (samholmes) [reply src/util/exchangeHelpers.js:3]: I suggest we create a patch/exchange-cdn branch and PR for the exchange icon CDN work. We can then make this PR and https://github.com/EdgeApp/edge-react-gui/pull/3296 dependent on it.

File: src/plugins/stake-plugins/policies/cemeteryPolicy.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (samholmes) [reply src/plugins/stake-plugins/policies/cemeteryPolicy.js:102]: Discovered: If you enter the same two token addresses into in different order, it doesn't effect the outcome of the contract call. As long as the amount parameters match the order of the given token address parameters, we're okay.

**Review by Jon-edge — APPROVED** [review]
gj

**Pull request merged**